torch/_dynamo/eval_frame.py

```diff
@@ -203,6 +205,24 @@ def get_example_inputs(key: str) -> list[Any]:
     return _EXAMPLE_INPUTS[key]
 
 
+@contextlib.contextmanager
+def _set_in_compiled_region():
```

Comment:
Can we add more documentation on how this differs from _compiling_state_context()/torch.compiler.is_compiling()?